app/silhouette/CustomVKProvider.scala

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package silhouette
+
+import com.mohiva.play.silhouette.api.util.HTTPLayer
+import com.mohiva.play.silhouette.impl.providers._
+import models.user.User
+import play.api.libs.json.JsValue
+
+import scala.concurrent.{ExecutionContext, Future}
+import com.mohiva.play.silhouette.impl.providers.oauth2.BaseVKProvider
+import com.mohiva.play.silhouette.impl.providers.oauth2.VKProfileParser
+
+/**
+  * Custom vk profile parser.
+  */
+class CustomVKProfileParser(implicit ec: ExecutionContext) extends CustomSocialProfileParser[JsValue, OAuth2Info] {
+
+  val parser = new VKProfileParser
+
+  def parse(content: JsValue, authInfo: OAuth2Info): Future[CustomSocialProfile] = {
+    parser.parse(content, authInfo).map { commonProfile =>
+      val gender = (content \ "sex").asOpt[Int].flatMap {
+        case 1 => Some(User.Gender.Male)
+        case 2 => Some(User.Gender.Female)
+        case _ => None
+      }
+
+      CustomSocialProfile(
+        loginInfo = commonProfile.loginInfo,
+        fullName = (commonProfile.firstName, commonProfile.lastName) match {
```

Comment:
I understand that this trivial yet tedious full name building method from optional first and last names is pretty much copypaste from external authentication builder so it arguably deserves refactor into a dedicated method/function, but I am asking to keep them both this way for now, because we plan to decouple full name into first name and last name in our model as well: https://bwsoft.atlassian.net/browse/SOP-494
cc: @elenaustyugova